flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/savepoint/SavepointV2Serializer.java

```diff
@@ -391,7 +392,7 @@ private static KeyedStateHandle deserializeKeyedStateHandle(DataInputStream dis)
 			Map<StateHandleID, StreamStateHandle> privateStates = deserializeStreamStateHandleMap(dis);
 
 			return new IncrementalKeyedStateHandle(
-				operatorId,
+				UUID.fromString(operatorId),
```

Comment:
This is not an operator ID anymore, right? Probably the local variable name should change.